spec/graphql/execution/breadth_runtime_spec.rb

```diff
@@ -0,0 +1,325 @@
+# frozen_string_literal: true
+require "spec_helper"
+
+describe "GraphQL::Execution::Interpreter for breadth-first execution" do
+  # A breadth-first interpreter uses the following runtime interface:
+  # - evaluate_selection(result_key, ast_nodes, selections_result)
+  # - exit_with_inner_result?
+  class SimpleBreadthRuntime < GraphQL::Execution::Interpreter::Runtime
+    class BreadthObject < GraphQL::Execution::Interpreter::Runtime::GraphQLResultHash
+      attr_accessor :breadth_index
+    end
+
+    def initialize(query:)
+      query.multiplex = GraphQL::Execution::Multiplex.new(
+        schema: query.schema,
+        queries: [query],
+        context: query.context,
+        max_complexity: nil,
+      )
+
+      super(query: query, lazies_at_depth: Hash.new { |h, k| h[k] = [] })
+      @breadth_results_by_key = {}
+    end
+
+    def run
+      result = nil
+      query.current_trace.execute_multiplex(multiplex: query.multiplex) do
+        query.current_trace.execute_query(query: query) do
+          result = yield
+        end
+      end
+      result
+    ensure
+      delete_all_interpreter_context
+    end
+
+    def evaluate_breadth_selection(objects, parent_type, node)
+      result_key = node.alias || node.name
+      @breadth_results_by_key[result_key] = Array.new(objects.size)
```

Comment:
In the actual prototype, breadth results are keyed with a unique representation of a field path.